api_types.ts

```diff
@@ -805,6 +807,65 @@ export interface Logger {
   error(message: string, ...args: LoggerParamType[]): void;
 }
 
+export enum InvocationErrorType {
+  Timeout = 'Timeout',
+
+  ResponseTooLarge = 'ResponseTooLarge',
+
+  HttpStatusError = 'HttpStatusError',
+
+  /**
+   * Could mean 3rd party API rate limit or a rate limit imposed by Coda.
+   */
+  RateLimitExceeded = 'RateLimitExceeded',
+
+  Unknown = 'Unknown',
+}
+
+interface BaseInvocationError {
+  type: InvocationErrorType;
```

Comment:
Do we want to start with only the error type? Worried pack authors will rely on message/errorData which feel unstable. 

Fine keeping all the information as well

Reply:
Yeah +1, I don't want an SDK contract where people rely on error text being stable, would prefer to only commit to the types.

Reply:
Makes sense! Removed both message & data

Reply:
Fwiw, I think including the actual failure when the failure was a RateLimit failure would be very useful for packs. 

Google for example includes metadata on when the rate limit would be over and it is safe to start making api calls. Unfortunately from the looks of it any api calls before that time is up continue to fail, and not only do they fail they cause the rate limit expiration to be pushed out.

So if the pack doesn't know what was the response from Google, the pack will have no choice but to re-issue the api calls and that will back fire and will cause the rate limit period to be extended. The only way this will get fixed is if the exponential backoff in our infra backs off far away enough to jump past the deadline given to us by google. This is very inefficient.

Here is an ingestion execution where this happened - https://admin.coda.io/ingestionExecutions/3db6cb4a-1662-4f96-8825-884bc1f697ff and if you look at the consecutive failures we failed 16 times in a row and every failure is causing the retry after time to get pushed out.

![image](https://github.com/user-attachments/assets/9bd0e865-01f4-4459-bdce-cda3c1c9ccde)

Reply:
So I think we should pass the raw failures back from the api calls back to the pack. Yes there is a risk that a pack may rely on some text that they shouldn't but that is up to the pack. We shouldn't penalize packs who know what they are doing because some packs may not know what they should do.

Reply:
For this case, we can probably read from the `retry-after` header and pass that in... although really in that case, we should just not even try to invoke the pack until that time is up. So I don't see much point in adding that additional data?